tmk_core/common/report.h

```diff
@@ -287,6 +290,11 @@ static inline uint16_t KEYCODE2CONSUMER(uint8_t key) {
     }
 }
 
+/* keycode to programmable buttons usage */
+static inline uint16_t KEYCODE2PROGRAMMABLEBUTTON(uint8_t key) {
+	return key - KC_PROGRAMMABLE_BUTTON0;
```

Comment:
This value no longer matches the usage code — for the first programmable button the value would be 0, but the HID usage code for that button is 1. Your `programmable_button_on()` and `programmable_button_off()` functions also expect such zero-base codes, but there is some potential for confusion, because the similar looking API for system control and consumer usages uses codes which are equal to HID usage codes.

Maybe it would be better to keep the offset of 1 in the API, at the cost of slightly more code inside `programmable_button_on()` and `programmable_button_off()`.

Reply:
Absolutely, will do.